docs/source/learn/trace-vs-execution-time.md

```diff
@@ -0,0 +1,175 @@
+# Tracing Time vs. Execution Time in PyTorch/XLA
+
+When working with PyTorch/XLA, it's essential to understand that operations on
+XLA tensors are not typically executed immediately in the way they are with
+standard PyTorch tensors on CPU or CUDA devices (which operate in "eager mode").
+PyTorch/XLA employs a "lazy execution" model. This means that when you write
+PyTorch code using XLA tensors, you are primarily defining or tracing a
+computation graph. The actual execution of this graph on the XLA device (like a
+TPU) is deferred until a specific trigger point.
+
+This leads to two distinct types of "time" to consider:
+
+1. Tracing Time: The period during which PyTorch/XLA records your operations and
+builds this graph on the host (CPU).
+
+1. Device Time: The period during which the XLA device (e.g., TPU) actually
+performs the computations. This can be further broken down:
+
+   - Compilation Time: The time XLA takes to compile the traced graph into
```

Comment:
The compilation runs on the host (inside the Python interpreter). It's not part of the device time. It _may_ be considered part of the tracing time, since `torch_xla.sync()` will synchronously block on the compilation.

Reply:
That is a good point! I have rephrased the host time to include tracing and cross compilation. With that execution is only attributed to device. PTAL